modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/ProjectGenerator.java

```diff
@@ -0,0 +1,245 @@
+		archetypeGenerationRequest.setInteractiveMode(false);
+		archetypeGenerationRequest.setOutputDirectory(destinationDir.getPath());
+		archetypeGenerationRequest.setPackage(packageName);
+
+		String buildType = "gradle";
+
+		if (projectTemplatesArgs.isMaven()) {
+			buildType = "maven";
+		}
+
+		Properties properties = new Properties();
+
+		_setProperty(properties, "author", author);
+		_setProperty(properties, "buildType", buildType);
+		_setProperty(properties, "className", className);
+		_setProperty(
+			properties, "dependencyManagementEnabled",
+			String.valueOf(dependencyManagementEnabled));
+		_setProperty(properties, "liferayVersion", liferayVersion);
+		_setProperty(properties, "package", packageName);
+		_setProperty(properties, "projectType", projectType);
+
+		archetypeGenerationRequest.setProperties(properties);
+
+		archetypeGenerationRequest.setVersion("1.0.0");
+
+		Archetyper archetyper = new Archetyper(archetypesDirs);
+
+		ArchetypeArtifactManager archetypeArtifactManager =
+			archetyper.createArchetypeArtifactManager();
+
+		ProjectTemplateCustomizer projectTemplateCustomizer =
+			_getProjectTemplateCustomizer(
+				archetypeArtifactManager.getArchetypeFile(
+					archetypeGenerationRequest.getArchetypeGroupId(),
+					archetypeGenerationRequest.getArchetypeArtifactId(),
+					archetypeGenerationRequest.getArchetypeVersion(), null,
+					null, null));
+
+		if (projectTemplateCustomizer != null) {
+			projectTemplateCustomizer.onBeforeGenerateProject(
+				projectTemplatesArgs, archetypeGenerationRequest);
+		}
+
+		ArchetypeManager archetypeManager = archetyper.createArchetypeManager();
+
+		ArchetypeGenerationResult archetypeGenerationResult =
+			archetypeManager.generateProjectFromArchetype(
+				archetypeGenerationRequest);
+
+		if (projectTemplateCustomizer != null) {
+			projectTemplateCustomizer.onAfterGenerateProject(
+				projectTemplatesArgs, destinationDir,
+				archetypeGenerationResult);
+		}
+
+		return archetypeGenerationResult;
+	}
+
+	private static File _getTemplateFile(
+			ProjectTemplatesArgs projectTemplatesArgs)
+		throws Exception {
+
+		String template = projectTemplatesArgs.getTemplate();
+
+		for (File archetypesDir : projectTemplatesArgs.getArchetypesDirs()) {
+			if (!archetypesDir.isDirectory()) {
+				continue;
+			}
+
+			try (DirectoryStream<Path> directoryStream =
+					Files.newDirectoryStream(
+						archetypesDir.toPath(), "*.project.templates.*")) {
+
+				for (Path path : directoryStream) {
+					String fileName = String.valueOf(path.getFileName());
+
+					String templateName = ProjectTemplatesUtil.getTemplateName(
+						fileName);
+
+					if (templateName.equals(template)) {
+						return path.toFile();
+					}
+				}
+			}
+		}
+
+		File archetypesFile = FileUtil.getJarFile(ProjectTemplates.class);
+
+		String artifactId =
+			ProjectTemplates.TEMPLATE_BUNDLE_PREFIX +
+				template.replace('-', '.');
+
+		return _getArchetypeFile(artifactId, archetypesFile);
```

Comment:
There's a compile error due to a missing method.

If it needs the same code in `ArchetyperArchetypeArtifactManager#_getArchetypeFile`, maybe it can be moved to one of the `internal/util` classes (i.e. `ProjectTemplatesUtil`)